src/main/scala/com/mongodb/spark/rdd/partitioner/MongoSinglePartitioner.scala

```diff
@@ -29,7 +29,21 @@ import com.mongodb.spark.config.ReadConfig
  *
  * @since 1.0
  */
-case object MongoSinglePartitioner extends MongoPartitioner {
-  override def partitions(connector: MongoConnector, readConfig: ReadConfig): Array[MongoPartition] =
-    Array(MongoPartition(0, new BsonDocument()))
+class MongoSinglePartitioner extends MongoPartitioner {
+
+  private var _partitions: Option[Array[MongoPartition]] = None
```

Comment:
I'm not clear why a single partitioner has a partitions array.   And where would it get set?